File: app/src/main/java/com/stalkstock/vender/adapter/MessageAdapter.java
package com.stalkstock.vender.adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.live.stalkstockcommercial.ui.view.activities.Chat;
import com.stalkstock.R;
import com.stalkstock.vender.Model.MessageList;
import com.stalkstock.vender.ui.ChatBox;

import java.util.List;


public class MessageAdapter extends RecyclerView.Adapter<MessageAdapter.ViewHolder> {

    int[] userimg ={R.drawable.chat_img, R.drawable.chat_img_2,R.drawable.chat_img_3,R.drawable.chat_img,R.drawable.chat_img_3, R.drawable.chat_img_2,R.drawable.chat_img, R.drawable.chat_img};
    Context context;
    LayoutInflater inflater;
    private List<MessageList> messageContacts;

    public MessageAdapter(List<MessageList> messageContacts) {
        this.messageContacts = messageContacts;
    }

    public  MessageAdapter(Context context){
        this.context=context;
        inflater = LayoutInflater.from(context);
    }

    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.messagelist, parent, false);
        ViewHolder viewHolder = new ViewHolder(view);

        viewHolder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                //context.startActivity(new Intent(context, ChatBox.class));
                context.startActivity(new Intent(context, Chat.class));
            }
        });
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
//        MessageList messageList = messageContacts.get(position);
//        holder.imageview.setImageResource(messageList.getImage());
//        holder.textname.setText(messageList.getName());
//        holder.texttime.setText(messageList.getTime());
//        holder.textView.setText(messageList.getMessage());

    }

    @Override
    public int getItemCount() {
        return 3;
    }

    public class ViewHolder extends RecyclerView.ViewHolder  {
        ImageView imageview;
        TextView textname,texttime,textView;
        public ViewHolder(View itemView) {
            super(itemView);
             imageview= itemView.findViewById(R.id.imguser);
            // textname=itemView.findViewById(R.id.msgusername);
             texttime=itemView.findViewById(R.id.messagetime);
          //  textView=itemView.findViewById(R.id.messagerecive);

        }


    }
}

File: app/src/main/java/com/stalkstock/vender/adapter/MessageAdapter.kt
package com.stalkstock.vender.adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.stalkstock.R
import com.stalkstock.commercial.view.activities.Chat
import com.stalkstock.utils.others.AppUtils
import com.stalkstock.utils.others.GlobalVariables
import com.stalkstock.utils.others.getPrefrence
import com.stalkstock.vender.Model.MessageList
import kotlinx.android.synthetic.main.messagelist.view.*

class MessageAdapter(
    var mContext: Context,
    var arrayList: ArrayList<MessageList>
) :
    RecyclerView.Adapter<MessageAdapter.RecyclerViewHolder>() {

    var receiverId=""

    var inflater: LayoutInflater

    class RecyclerViewHolder(view: View) : RecyclerView.ViewHolder(view) {
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerViewHolder {
        val v = inflater.inflate(R.layout.messagelist, parent, false)
        return RecyclerViewHolder(v)
    }

    override fun onBindViewHolder(holder: RecyclerViewHolder, position: Int) {

        if(arrayList[position].senderId==getPrefrence(GlobalVariables.SHARED_PREF_DRIVER.id, 0)){
            holder.itemView.tv_personName.text= arrayList[position].receiver.firstName + " "+arrayList[position].receiver.lastName
            Glide.with(mContext).load(arrayList[position].receiver.image).into(holder.itemView.iv_userimage)

        }else{
            holder.itemView.tv_personName.text= arrayList[position].sender.firstName + " "+arrayList[position].sender.lastName
            Glide.with(mContext).load(arrayList[position].sender.image).into(holder.itemView.iv_userimage)

        }


        holder.itemView.tv_messageTemplate.text= arrayList[position].lastMessage.message
        holder.itemView.tv_timeOFMessage.text= AppUtils.convertTimeStampToDateTime(arrayList[position].created.toLong())

        holder.itemView.setOnClickListener {

            val intent = Intent(mContext, Chat::class.java)
            if (arrayList[position].receiverId == getPrefrence(GlobalVariables.SHARED_PREF_DRIVER.id, 0)){
                intent.putExtra("userId",arrayList[position].senderId)
                intent.putExtra("userName", arrayList[position].sender.firstName + " "+arrayList[position].sender.lastName)
                intent.putExtra("userImage",arrayList[position].sender.image.toString())

            }else{
                intent.putExtra("userImage",arrayList[position].receiver.image.toString())
                intent.putExtra("userName", arrayList[position].receiver.firstName + " "+arrayList[position].receiver.lastName)


                intent.putExtra("userId",arrayList[position].receiverId)
            }

            if(arrayList[position].bidId==0){
                intent.putExtra("id",arrayList[position].orderId.toString())
                intent.putExtra("param_name","orderId")
            }else{
                intent.putExtra("id",arrayList[position].bidId.toString())
                intent.putExtra("param_name","bidId")
            }

            intent.putExtra("chatId",arrayList[position].id.toString())
            if (arrayList[position].receiverId == getPrefrence(GlobalVariables.SHARED_PREF_DRIVER.id, 0)){
                intent.putExtra("userId",arrayList[position].senderId)

            }else{
                intent.putExtra("userId",arrayList[position].receiverId)
            }
            mContext.startActivity(intent)

        }

    }

    override fun getItemCount(): Int {
        return arrayList.size
    }

    init {
        inflater = LayoutInflater.from(mContext)
    }
}
